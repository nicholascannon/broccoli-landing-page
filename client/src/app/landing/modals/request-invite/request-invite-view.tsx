import { useState } from 'react';
import { Button } from '../../../../common/components/button';
import { FormInput } from '../../../../common/form/form-input';
import { GenericModalView } from '../../../../common/modals/components/generic-modal-view';

import styles from './request-invite-view.module.css';

export const RequestInviteModalView = (props: ViewProps) => {
    const { onRequestInvite, loading, serverError, formErrors } = props;

    const [name, setName] = useState('');
    const [email, setEmail] = useState('');
    const [confirmEmail, setConfirmEmail] = useState('');

    const onSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
        event.preventDefault();
        onRequestInvite(name, email, confirmEmail);
    };

    return (
        <GenericModalView>
            <div className={styles.content}>
                <div>
                    <h2>Request an invite</h2>
                    <hr />
                </div>

                <form className={styles.requestForm} onSubmit={onSubmit} noValidate>
                    <FormInput
                        type="text"
                        name="name"
                        id="name"
                        placeholder="Full name"
                        value={name}
                        onChange={(e) => setName(e.target.value.trim())}
                        required
                        readOnly={loading}
                        error={formErrors?.name}
                    />

                    <FormInput
                        type="email"
                        name="email"
                        id="email"
                        placeholder="Email"
                        value={email}
                        onChange={(e) => setEmail(e.target.value.trim())}
                        required
                        readOnly={loading}
                        error={formErrors?.email}
                    />

                    <FormInput
                        type="email"
                        name="confirm-email"
                        id="confirm-email"
                        placeholder="Confirm email"
                        value={confirmEmail}
                        onChange={(e) => setConfirmEmail(e.target.value.trim())}
                        required
                        readOnly={loading}
                        error={formErrors?.confirmEmail}
                    />

                    <Button fullWidth type="submit" className={styles.callToAction} disabled={loading}>
                        {loading ? 'Sending, please wait...' : 'Send'}
                    </Button>
                </form>

                {serverError && <span className={styles.serverError}>{serverError}</span>}
            </div>
        </GenericModalView>
    );
};

type ViewProps = {
    loading: boolean;
    serverError?: string;
    formErrors?: {
        name?: string;
        email?: string;
        confirmEmail?: string;
    };
    onRequestInvite: (name: string, email: string, confirmEmail: string) => void;
};
