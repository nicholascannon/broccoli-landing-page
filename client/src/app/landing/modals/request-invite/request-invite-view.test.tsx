/* eslint-disable @typescript-eslint/no-empty-function */
import renderer from 'react-test-renderer';
import { RequestInviteModalView } from './request-invite-view';

test(`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there are no errors
    THEN rendering it should match the snapshot
`, () => {
    const tree = renderer.create(<RequestInviteModalView loading={false} onRequestInvite={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test(`
    GIVEN RequestInviteModalView component
    WHEN loading is true and there are no errors
    THEN rendering it should match the snapshot
`, () => {
    const tree = renderer.create(<RequestInviteModalView loading={true} onRequestInvite={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test(`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there is a server error
    THEN rendering it should match the snapshot
`, () => {
    const tree = renderer
        .create(<RequestInviteModalView loading={false} onRequestInvite={() => {}} serverError="Error" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test(`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there is form errors
    THEN rendering it should match the snapshot
`, () => {
    const tree = renderer
        .create(
            <RequestInviteModalView
                loading={false}
                onRequestInvite={() => {}}
                formErrors={{
                    name: 'Error',
                    email: 'Error',
                    confirmEmail: 'Error',
                }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
