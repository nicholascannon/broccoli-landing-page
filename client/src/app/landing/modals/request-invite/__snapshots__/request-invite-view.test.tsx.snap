// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there are no errors
    THEN rendering it should match the snapshot
 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <div>
      <h2>
        Request an invite
      </h2>
      <hr
        className="ruler ruler"
      />
    </div>
    <form
      className="requestForm"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <input
          className="input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Full name"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="confirm-email"
          name="confirm-email"
          onChange={[Function]}
          placeholder="Confirm email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <button
        className="button fullWidth callToAction"
        disabled={false}
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;

exports[`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there is a server error
    THEN rendering it should match the snapshot
 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <div>
      <h2>
        Request an invite
      </h2>
      <hr
        className="ruler ruler"
      />
    </div>
    <form
      className="requestForm"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <input
          className="input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Full name"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="confirm-email"
          name="confirm-email"
          onChange={[Function]}
          placeholder="Confirm email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <button
        className="button fullWidth callToAction"
        disabled={false}
        type="submit"
      >
        Send
      </button>
    </form>
    <span
      className="serverError"
    >
      Error
    </span>
  </div>
</div>
`;

exports[`
    GIVEN RequestInviteModalView component
    WHEN loading is false and there is form errors
    THEN rendering it should match the snapshot
 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <div>
      <h2>
        Request an invite
      </h2>
      <hr
        className="ruler ruler"
      />
    </div>
    <form
      className="requestForm"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <input
          className="input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Full name"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="errorVisible"
        >
          <span
            className="errorMessage"
          >
            Error
          </span>
        </div>
      </div>
      <div>
        <input
          className="input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorVisible"
        >
          <span
            className="errorMessage"
          >
            Error
          </span>
        </div>
      </div>
      <div>
        <input
          className="input"
          id="confirm-email"
          name="confirm-email"
          onChange={[Function]}
          placeholder="Confirm email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorVisible"
        >
          <span
            className="errorMessage"
          >
            Error
          </span>
        </div>
      </div>
      <button
        className="button fullWidth callToAction"
        disabled={false}
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;

exports[`
    GIVEN RequestInviteModalView component
    WHEN loading is true and there are no errors
    THEN rendering it should match the snapshot
 1`] = `
<div
  className="modal"
>
  <div
    className="content"
  >
    <div>
      <h2>
        Request an invite
      </h2>
      <hr
        className="ruler ruler"
      />
    </div>
    <form
      className="requestForm"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <input
          className="input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Full name"
          readOnly={true}
          required={true}
          type="text"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          readOnly={true}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <div>
        <input
          className="input"
          id="confirm-email"
          name="confirm-email"
          onChange={[Function]}
          placeholder="Confirm email"
          readOnly={true}
          required={true}
          type="email"
          value=""
        />
        <div
          className="errorHidden"
        >
          <span
            className="errorMessage"
          />
        </div>
      </div>
      <button
        className="button fullWidth callToAction"
        disabled={true}
        type="submit"
      >
        Sending, please wait...
      </button>
    </form>
  </div>
</div>
`;
